testsuite/MDAnalysisTests/analysis/test_base.py

```diff
@@ -187,6 +94,109 @@ def u_xtc():
+    (SerialOnly, 'not-existing-backend', None),
+    (SerialOnly, 'multiprocessing', 2),
+    (SerialOnly, 'dask', None),
+    (ParallelizableWithDaskOnly, 'multiprocessing', None),
+    (ParallelizableWithDaskOnly, 'multiprocessing', 2),
+])
+def test_backend_configuration_fails(u, run_class, backend, n_workers):
+    u = mda.Universe(TPR, XTC)  # dt = 100
+    with pytest.raises(ValueError):
+        _ = run_class(u.trajectory).run(backend=backend, n_workers=n_workers, stop=0)
+
+@pytest.mark.parametrize('run_class,backend,n_workers', [
+    (Parallelizable, CustomSerialBackend, 2),
+    (ParallelizableWithDaskOnly, CustomSerialBackend, 2),
+])
+def test_backend_configuration_works_when_unsupported_backend(u, run_class, backend, n_workers):
+    u = mda.Universe(TPR, XTC)  # dt = 100
+    backend_instance = backend(n_workers=n_workers)
+    _ = run_class(u.trajectory).run(backend=backend_instance, n_workers=n_workers, stop=0, unsupported_backend=True)
+
+@pytest.mark.parametrize('run_class,backend,n_workers', [
+    (Parallelizable, CustomSerialBackend, 1),
+    (ParallelizableWithDaskOnly, CustomSerialBackend, 1),
+])
+def test_custom_backend_works(u, run_class, backend, n_workers):
+    backend_instance = backend(n_workers=n_workers)
+    u = mda.Universe(TPR, XTC)  # dt = 100
+    _ = run_class(u.trajectory).run(backend=backend_instance, n_workers=n_workers, unsupported_backend=True)
+
+@pytest.mark.parametrize('run_class,backend_instance,n_workers', [
+    (Parallelizable, map, 1),
+    (SerialOnly, list, 1),
+    (ParallelizableWithDaskOnly, object, 1),
+])
+def test_fails_incorrect_custom_backend(u, run_class, backend_instance, n_workers):
+    u = mda.Universe(TPR, XTC)  # dt = 100
+    with pytest.raises(ValueError):
+        _ = run_class(u.trajectory).run(backend=backend_instance, n_workers=n_workers, unsupported_backend=True)
+
+    with pytest.raises(ValueError):
+        _ = run_class(u.trajectory).run(backend=backend_instance, n_workers=n_workers)
+
+@pytest.mark.parametrize('run_class,backend,n_workers', [
+    (SerialOnly, CustomSerialBackend, 1),
+    (SerialOnly, 'multiprocessing', 1),
+    (SerialOnly, 'dask', 1),
+])
+def test_fails_for_unparallelizable(u, run_class, backend, n_workers):
+    u = mda.Universe(TPR, XTC)  # dt = 100
+    with pytest.raises(ValueError):
+        if not isinstance(backend, str):
+            backend_instance = backend(n_workers=n_workers)
+            _ = run_class(u.trajectory).run(backend=backend_instance, n_workers=n_workers, unsupported_backend=True)
+        else:
+            _ = run_class(u.trajectory).run(backend=backend, n_workers=n_workers, unsupported_backend=True)
+
+@pytest.mark.parametrize('run_kwargs,frames', [
+    ({}, np.arange(98)),
+    ({'start': 20}, np.arange(20, 98)),
+    ({'stop': 30}, np.arange(30)),
+    ({'step': 10}, np.arange(0, 98, 10))
+])
+def test_start_stop_step_parallel(u, run_kwargs, frames, client_FrameAnalysis):
```

Comment:
Add a comment describing where `client_FrameAnalysis` is defined and what it does.

Global fixture definitions in pytest seem necessary but they make for very unreadable code — too much magic going on. Thus, comments are necessary.

Reply:
Done